domain/ide/workspace/workspace.go

```diff
@@ -62,10 +62,21 @@ func Set(w *Workspace) {
 	instance = w
 }
 
-func (w *Workspace) DeleteFolder(folder string) {
+func (w *Workspace) DeleteFolder(folderPath string) {
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
-	delete(w.folders, folder)
+	folder := w.GetFolderContaining(folderPath)
+	folder.ClearDiagnosticsFromPathRecursively(folderPath)
+	delete(w.folders, folderPath)
+}
+
+func (w *Workspace) DeleteFile(filePath string) {
+	w.mutex.Lock()
+	defer w.mutex.Unlock()
+	folder := w.GetFolderContaining(filePath)
+	if folder != nil {
+		folder.ClearDiagnosticsFromFile(filePath)
```

Comment:
This method resets folder scan status. Should we reset it when a file was deleted? What is the implication of that? 

Also `server.go` resets scan status for a folder (though marked as no-op).

Reply:
Clearing scanned status is preserving existing behavior. Not sure about the implications either but I didn't want to change it.

Reply:
Actually, it looks like the scanned status is only used in tests 🤣

Reply:
Oh, okay :)